packages/eslint-plugin/docs/rules/consistent-return.md

```diff
@@ -0,0 +1,32 @@
+---
+description: 'Require `return` statements to either always or never specify values.'
+---
+
+> 🛑 This file is source code, not the primary documentation location! 🛑
+>
+> See **https://typescript-eslint.io/rules/consistent-return** for documentation.
+
+This rule extends the base [`eslint/consistent-return`](https://eslint.org/docs/rules/consistent-return) rule.
+This version adds support for functions that return void type.
```

Comment:
[Docs] Adding a touch of syntax highlighting, and also being explicit that void Promises are supported:

```suggestion
This version adds support for functions that return `void` or `Promise<void>`.
```